src/object/environment.rs

```diff
@@ -45,31 +54,70 @@ impl Environment {
     }
 
     pub fn get(&self, name: String) -> EvalResult {
-        // search in this environment for value by name
+        self.get_internal(name, 0).0
+    }
+
+    fn get_internal(&self, name: String, i: u32) -> (EvalResult, i32) {
```

Comment:
From what I can tell, aside from recursive calls `i` is always provided a default of 0 by external calls. Likewise, the `i32` that's returned seems to only be used for tracking whether the value is returned from the current frame. 

Since the parameter `i` is seems to only be used for tracking recursion, this looks like a good candidate for rewriting the recursive behavior as a loop.

I also think the returned `i32` seems like it could be handled by a `bool` indicating whether the value was returned by the current frame.

Let me know if I'm missing something and there's some other purpose that these are needed for.

Reply:
Yes, this makes a lot of sense, thanks for the review already and the great feedback!